CONTRIBUTING.md

```diff
@@ -0,0 +1,66 @@
+### TLDR;
+
+Before doing anything else, run `cargo build -p lalrpop`!
+
+From now on compile/test with `cargo compile/test -p lalrpop --features=test`.
+
+Just before your final commit, run `sh update_lrgrammar.sh`
+
+
+### Contributor's "How to"
+
+Just as any language that is written in itself, LALRPOP has to solve chicken-and-egg problem.
+To compile LALRPOP we need a working version of LALRPOP. To have a working version we need
+to compile it first.
+
+In the past, LALRPOP used snapshots of its older self to compile itself (you might notice
+[lalrpop-snap](https://crates.io/crates/lalrpop-snap) on crates.io). Unfortunately, this meant end-users
+had to build **both** `lalrpop` and `lalrpop-snap` crates and nobody liked this double-building.
+This approach has been abandoned a while ago.
+
+Today's approach is different - `lalrpop` crate includes already auto-generated parser
+in `lalrpop/src/parser/lrgrammar.rs` that's end-users use.
+Small (meh), (relatively) fast to compile, easy to use (really, for end-users).
+
+That said, if you changes don't affect LALRPOP's own grammar
+(`lalrpop/src/parser/lrgrammar.lalrpop`) your workflow is simple
+
+```sh
+// building
+$ cargo build // -p lalrpop --release
+
+// testing
+$ cargo test // -p lalrpop --release
```

Comment:
Should this maybe use `#` for shell comments?

Reply:
Right, will do